src/main/java/org/gradle/profiler/mutations/support/ScenarioSupport.java

```diff
@@ -0,0 +1,33 @@
+package org.gradle.profiler.mutations.support;
+
+import com.typesafe.config.Config;
+import org.gradle.profiler.ConfigUtil;
+
+import java.io.File;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ScenarioSupport {
+
+    public static List<File> sourceFiles(Config config, String scenarioName, File projectDir, String key) {
+        return ConfigUtil.strings(config, key)
+            .stream()
+            .map(fileName -> openFile(fileName, projectDir, scenarioName))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
+
+    private static File openFile(String fileName, File projectDir, String scenarioName) {
+        if (fileName == null) {
+            return null;
+        } else {
+            File file = new File(projectDir, fileName);
+            if (!file.isFile()) {
+                throw new IllegalArgumentException("Source file " + file.getName() + " specified for scenario " + scenarioName + " does not exist.");
+            }
+            return file;
+        }
+    }
+
```

Comment:
How about something like this?

```java
    public static List<File> sourceFiles(Config config, String scenarioName, File projectDir, String key) {
        return ConfigUtil.strings(config, key)
            .stream()
            .map(fileName -> new File(projectDir, fileName))
            .peek(file -> {
                if (!file.isFile()) {
                    throw new IllegalArgumentException("Source file " + file.getName() + " specified for scenario " + scenarioName + " does not exist.");
                }
            })
            .collect(Collectors.toList());
    }
```

I understand that you didn't write this code, but I need to write my self-assessment and needed distraction. Sorry. :)

Reply:
We could probably also return the stream instead of converting to a `List`, too.

Reply:
Used that code and also add `.filter(Objects::nonNull)` since I am not sure if ConfigUtil can return null. Better safe than sorry